[team6_pair1] 문자열계산기, 레이싱 게임 구현 완료 step1

kwangilcho opened the pull request.

Commit:
Implement Calculator class (#12)

Commit:
Step0 calculator 클래스를 완성하였습니다.

1. caculator 클래스는 한 줄 입력을 받아 계산해주는 클래스입니다.
2. plus, minus, multiply, divide, 그리고 입력에 대한 에러 테스트를 진행하였습니다.

Commit:
자동차 경주 게임 구현 완료

javajigi commented on src/main/java/racingcar/Car.java:
```diff
@@ -0,0 +1,22 @@
+public class Car {
+    private int position = 0;
+    public void move(int randomNo) {
+        if (randomNo > 3) {
```
값을 하드코딩하지 마라.
상수(static final)를 만들고 이름을 부여해 이 변수의 역할이 무엇인지 의도를 드러내라.

javajigi commented on src/main/java/racingcar/Car.java:
```diff
@@ -0,0 +1,22 @@
+    public String repeat() {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < this.position; i++) {
+            sb.append("-");
```
값을 하드코딩하지 마라.
상수(static final)를 만들고 이름을 부여해 이 변수의 역할이 무엇인지 의도를 드러내라.

javajigi commented on src/main/java/racingcar/Car.java:
```diff
@@ -0,0 +1,22 @@
+        return position;
+    }
+
+    public String repeat() {
```
인스턴스 변수에 의존하지 않아 클래스 메소드로 구현해도 되지 않을까요?
이와 비슷한 종류의 메소드를 찾아 클래스 메소드로 구현한다.

javajigi commented on src/main/java/racingcar/RacingCarGame.java:
```diff
@@ -0,0 +1,51 @@
+import java.util.ArrayList;
+
+public class RacingCarGame {
+    ArrayList<Car> cars = new ArrayList<>();
```
private으로 구현..
ArrayList<Car> => List<Car>로 구현한다.

javajigi commented on src/main/java/racingcar/RacingCarGame.java:
```diff
@@ -0,0 +1,51 @@
+public class RacingCarGame {
+    ArrayList<Car> cars = new ArrayList<>();
+
+    public void initCars(int numCar) {
```
이 메소드를 생성자에서 구현하는 것은 어떨까?

javajigi commented on src/test/java/racingcar/CarTest.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Test
+    public void carMoveSuccess() {
+        car.move(8);
```
이런 값보다는 4의 경계 값을 테스트하는 것을 추천.
즉, 3과 4가 좋은 값임

javajigi reviewed (changes requested):
구현 잘 했네요. 👍 
피드백 남겼어요.

Commit:
Step1 피드백 사항 모두 적용하였습니다.

- List Pointer with ArrayList instance
  https://stackoverflow.com/questions/2279030/type-list-vs-type-arraylist-in-java
- Do not hard code constant value

kwangilcho commented:
피드백 감사합니다!
주신 피드백 모두 반영하였습니다.
